shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Utils.hs

```diff
@@ -182,7 +183,8 @@ data RawSeed = RawSeed !Word64 !Word64 !Word64 !Word64 !Word64
 
 instance Arbitrary RawSeed where
   arbitrary =
-    RawSeed <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary
+    RawSeed
+      <$> chooseAny <*> chooseAny <*> chooseAny <*> chooseAny <*> chooseAny
```

Comment:
I'd love to learn from the `Random` maintainer about the when to use `arbitrary` vs `chooseAny`. :smile: (if it's long, you can inform me later, or give me a reading assignment)

Reply:
It is not long at all, but it is an extremely important differece! If you use `Random`/`Uniform` class then you'll get uniform distribution, but when you use `Arbitrary` instance then you get some sort of randomness that depends on `size` For example `chooseAny` for `Int` will pick a truly pseudo random number uniformly on the whole range `minBound .. maxBound`:
```haskell
λ> sample (chooseAny :: Gen Int)
8210185546773420831
518266691867196564
-6264935124352629180
1513173028515106965
6464989126909467446
-3226533972854067600
-2010886522517333871
-7398707487487345588
5672178984769180970
-5781351238750652131
244116361716631734
```
while `arbitrary` will give you small values that can be scaled with size:
```haskell
λ> sample (((,) <$> arbitrary <*> resize 300 arbitrary) :: Gen (Int, Int))
(0,-296)
(-2,156)
(0,-141)
(5,-153)
(-3,-147)
(-10,-242)
(7,-77)
(4,288)
(-12,167)
(-18,151)
(6,-80)
```

Latter is useful for testing, but sometimes it gives you a false sense of security because the range of values that is tested is not what most people expect, eg:

```haskell
λ> quickCheck $ forAll (arbitrary :: Gen Int) (< 100)
+++ OK, passed 100 tests.
```

Reply:
ah, makes sense, thanks!